---
title: "Conserved RNA structures (CRSs) in cyanobacteria"
---

This web resource presents **candidates for novel predicted conserved RNA structures (novel CRSs)** and **known RNA structures** inside intergenic regions flanking **931 orthologous protein-coding gene groups** in **202 cyanobacteria species-representative genomes** from the [proGenomes](https://progenomes.embl.de) database.

This large-scale, phylum-wide comparative genomics study screens the intergenic regions located adjacent up-/downstream of orthologous genes for  

1. bacterial known RNA structure families provided in [Rfam](https://rfam.org/) database and bacterial transcription terminators by homology search using [Infernal cmsearch](http://eddylab.org/infernal/), and  
2. novel CRSs using the [CMfinder](https://sourceforge.net/projects/weinberg-cmfinder/) tool and the local phylogeny of the orthologous genes.

The novel CRSs are characterized by  

- structure covariation (determined with the [R-scape](http://eddylab.org/R-scape/) tool) and sequence conservation of the [post-processed structure alignment](help.html),  
- putative *cis-*acting function through guilt by association to the annotated pathways of adjacent orthologous genes, and  
- transcriptional activity based on >400 processed public single-end RNA-seq libraries from 9 genomes (using the the zFPKM method).


```{r}
library(tidyverse)
library(kableExtra)

setwd("/media/dna/users/seemann/projects-git/OCyRS-website")

dat <-
  'data/potentially-novel-motifs.tsv' |>
  read_tsv() |>
  select(motif, category) |>
  # add pathway indication
  left_join(
    'data/K_motif-path.tsv' |>
      read_tsv() |> 
      drop_na(pathway) |>
      select(motif) |>
      unique() |>
      mutate(has.path = TRUE),
    'motif'
  ) |>
  # expressed genomes indication
  left_join(
    'data/5-maybe-interest.tsv' |>
      read_tsv(),
    c('motif' = 'gene')
  )

# Collect values for overview table
dat2 <- bind_rows(
  dat |>
    count(category) |>
    mutate(col = 'Candidate RNA structures'),
  dat |>
    filter(has.path) |>
    count(category) |>
    mutate(col = 'Associated with pathways'),
  dat |>
    filter(expressed.in.genomes >= 1) |>
    count(category) |>
    mutate(col = 'Transcriptionally active'),
  dat |>
    filter(expressed.in.genomes >= 2) |>
    count(category) |>
    mutate(col = 'Transcriptionally active in â‰¥ 2 genomes')
)

# Add total row
dat3 <- bind_rows(
  dat2,
  dat2 |>
    group_by(col) |>
    summarize(category = 'In total', n = sum(n))
)
# spread table, arrange rows etc
dat3 |>
  # keep order of columns
  mutate_at('col', fct_inorder) |>
  # spread out
  spread(col, n) |>
  rename(Category = category) |>
  # arrange rows
  mutate_at(
    'Category', fct_relevel,
    c(
      "High covariation", 
      "Conserved sequence (low covariation and power)",
      "High power (low covariation)",
      "In total"
    )) |>
  arrange(Category) |>
  # requested renaming
  mutate_at('Category', fct_recode,
            "High sequence conservation" = "Conserved sequence (low covariation and power)",
            "High power" = "High power (low covariation)") |>
  # print in html
  kable(caption = 'Counts of novel CRSs and their characterizations') %>%
  kable_paper('hover', full_width = FALSE) %>%
  row_spec(4, bold = TRUE)
```

Click [Browse CRSs](browse.html) to get started with exploring the novel CRSs.  
Click [Known structures](cmsearch.html) to explore the Rfam hits and transcription terminators inside the intergenic search regions of this computational screen.

The data can also be downloaded as flat files:  

- [List of genomic locations and taxonomy of the novel CRSs](data/K_motif-tax-pos.tsv)  
- [List of all novel CRSs and their associated pathways](data/K_motif-path.tsv)
- [Archive of the STOCKHOLM format alignments of all novel CRSs](data/crs-alignment.tar.gz)
- [Archive of the Covariance Models of all novel CRSs](data/crs-covariance_models.tar.gz)
- [List of Rfam families in the 202 genomes](data/G_rfam-cmsearch.tsv.gz)
- [List of terminator sequences in the 202 genomes](data/G2_terminators.tsv.gz)

If you use this resource, please cite:  

**Exploring the RNA structure regulatory potential in 202 cyanobacterial genomes**  
Adrian Sven Geissler, Elena Carrasquer-Alvarez, Christian Anthon, Niels-Ulrik Frigaard, Jan Gorodkin, and Stefan Ernst Seemann  
*submitted*
